src/app/routes/utils/regex/utils/index.js

```diff
@@ -2,7 +2,7 @@ const idRegex = 'c[a-zA-Z0-9]{10}o';
 const ampRegex = '.amp';
 const assetUriRegex = '[a-z-_]{0,}[0-9]{8,}';
 const legacyAssetUriRegex =
-  '[a-z-_]{1,}/[0-9]{4}/[0-9]{2}/[0-9]{6}[a-z0-9-_]{0,}';
+  '[a-z-_/]{1,}/[0-9]{4}/[0-9]{2}/[0-9]{6}[a-z0-9-_]{0,}';
```

Comment:
The current regex would match these routes:
```
/afrique/one/two/three/four/five/six/seven/---/infinitenumberofthese---/1900/01/123456-abcdefg
/zhongwen/simp/one/two/three/four/five/six/seven/---/infinitenumberofthese---/1900/01/123456-abcdefg

```

```suggestion
  '([a-z-_]{1,}/){1,5}[0-9]{4}/[0-9]{2}/[0-9]{6}[a-z0-9-_]{0,}';
```
In this suggestion, we're limiting this part to 5 groups prior to the date.

e.g. a maximum of:
```
/afrique/one/two/three/four/five/1900/01/123456-abcdefg
/zhongwen/simp/one/two/three/four/five/1900/01/123456-abcdefg
```
rather than allowing an infinite number of paths between the service/variant and the year in the URL.

Reply:
Ah, it seems that React Router doesn't like nested matches, since although wrapping the` [a-z-_/]{1,}/` in parentheses is valid regex (see https://regex101.com/), it doesn't match with React Router. 😞

Reply:
I think this is because adding the parentheses is creating a secondary capture group